numbuf/python/src/pynumbuf/numbuf.cc

```diff
@@ -170,14 +186,90 @@ static PyObject* register_callbacks(PyObject* self, PyObject* args) {
   return result;
 }
 
+#ifdef HAS_PLASMA
+
+#include "plasma_extension.h"
+
+static void BufferCapsule_Destructor(PyObject* capsule) {
+  object_id *id = reinterpret_cast<object_id*>(PyCapsule_GetPointer(capsule, "buffer"));
+  plasma_connection *conn = reinterpret_cast<plasma_connection*>(PyCapsule_GetContext(capsule));
+  plasma_release(conn, *id);
+  delete id;
+}
+
+static PyObject* store_list(PyObject* self, PyObject* args) {
+  object_id obj_id;
+  plasma_connection *conn;
+  PyObject *value;
+  if (!PyArg_ParseTuple(args, "O&O&O", PyObjectToUniqueID, &obj_id, PyObjectToPlasmaConnection, &conn, &value)) {
+    return NULL;
+  }
+  if (!PyList_Check(value)) {
+    return NULL;
+  }
+
+  std::shared_ptr<Array> array;
+  int32_t recursion_depth = 0;
+  Status s = SerializeSequences(std::vector<PyObject*>({value}), recursion_depth, &array);
+  CHECK_SERIALIZATION_ERROR(s);
+
+  auto batch = new std::shared_ptr<RecordBatch>();
+  std::shared_ptr<Buffer> metadata;
+  int64_t size = make_schema_and_batch(array, &metadata, batch);
```

Comment:
should be called `metadata_offset`, document what it is for and how the arrow data is being mapped into plasma data/metadata